src/_guides/libraries/testing.md

```diff
@@ -0,0 +1,130 @@
+  test app generally runs on a different device or OS.
+
+## Generally useful libraries
+
+How you write your tests partly depends on whether you are using Flutter,
+Dart VM, AngularDart, or other web solutions, but the following packages
+are useful across Dart platforms:
+
+* [package:test](https://pub.dartlang.org/packages/test)<br>
+  Provides a standard way of writing tests in Dart. You can use the test
+  package to:
+    * Write single tests, or groups of tests.
+    * Use the `@TestOn` annotation to restrict tests to run on
+      specific environments.
+    * Write asynchronous tests just as you would for non-asynchronous
+      tests.
+    * Tag tests using the `@Tag` annotation. For example, define a tag to
+      create a custom configuration for some tests, or to identify some tests
+      as needing more time to complete.
+    * Create a `dart_test.yaml` file to configure tagged tests across
+      multiple files or an entire package.
+
+
+* [package:mockito](https://pub.dartlang.org/packages/mockito)<br>
+  Provides a way to mock interfaces so that a piece of code is replaced
+  by a dummy implementation that emulates real code, but behaves in
+  a predictable way. This package is particularly useful for unit
+  testing. For an example that uses both package:test and package:mockito,
+  see the [International Space Station API library and its unit
+  tests](https://github.com/dart-lang/mockito/tree/master/test/example/iss)
+  in the [mockito package](https://github.com/dart-lang/mockito).
+
+## Flutter testing
+
+Use the following resources to learn more about testing Flutter apps:
+
+* [Testing Flutter Apps](https://flutter.io/testing/)<br>
+  How to perform unit, widget, or integration tests on a Flutter app.
+* [flutter_test](https://docs.flutter.io/flutter/flutter_test/flutter_test-library.html)<br>
+  A testing library for Flutter built on top of package:test.
+* [flutter_driver](https://docs.flutter.io/flutter/flutter_driver/flutter_driver-library.html)<br>
+  A testing library for testing Flutter applications on real devices and
+  emulators (in a separate process).
+* [flutter/examples/flutter_gallery](https://github.com/flutter/flutter/tree/master/examples/flutter_gallery)<br>
+  Tests for the Flutter gallery example.
+* [flutter/dev/manual_tests](https://github.com/flutter/flutter/tree/master/dev/manual_tests)<br>
+  Many examples of tests in the Flutter SDK.
+
+## Web testing
+
+Use the following resources to learn more about testing Dart web
+applications:
+
+* [Testing](https://webdev.dartlang.org/angular/guide/testing)<br>
+  How to use the [angular_test](https://pub.dartlang.org/packages/angular_test)
+  package to test AngularDart components and Angular subsystems.
+* [package:webdriver](https://pub.dartlang.org/packages/webdriver)<br>
+  Provides bindings that use the WebDriver JSON interface and require
+  the WebDriver remote server. [PENDING: a better way to describe this?]
+
+## Other tools and resources
```

Comment:
The best debugging experience is still the IDE. Perhaps mention that IntelliJ (and maybe some others?) has a full-featured Dart debugger?